lua/neo-tree/sources/filesystem/lib/filter_external.lua

```diff
@@ -43,133 +43,297 @@ local get_find_command = function(state)
   return state.find_command
 end
 
-M.find_files = function(opts)
-  local filters = opts.filtered_items
-  local limit = opts.limit or 200
-  local cmd = get_find_command(opts)
-  local path = opts.path
-  local full_path_words = opts.find_by_full_path_words
-  local regex, glob = opts.term, opts.term
-  local fuzzy_finder_mode = opts.fuzzy_finder_mode
+---@class FileTypes
+---@field file boolean
+---@field directory boolean
+---@field symlink boolean
+---@field socket boolean
+---@field pipe boolean
+---@field executable boolean
+---@field empty boolean
+---@field block boolean Only for `find`
+---@field character boolean Only for `find`
 
-  if full_path_words then
-    local words = utils.split(glob, " ")
-    regex = ".*" .. table.concat(words, ".*") .. ".*"
-  else
-    if glob ~= "*" then
-      if glob:sub(1) ~= "*" then
-        glob = "*" .. glob
-      end
-      if glob:sub(-1) ~= "*" then
-        glob = glob .. "*"
-      end
-    end
+---filter_files_external
+-- Spawns a filter command based on `cmd`
```

Comment:
```suggestion
```
I don't think there is any valid reason to force a particular program to be used here. It's an internal implementation detail that the consumer of this function should not need to know about.

Also, everyone has different programs available and we should leave it up to this function to choose the best available option.